src/compiler/crystal/semantic/ast.cr

```diff
@@ -77,6 +77,23 @@ module Crystal
     def_equals_and_hash type
   end
 
+  # Fictitious node to represent a type restriction
+  #
+  # It is used for type restrection of method arguments.
+  class TypeRestrict < ASTNode
+    getter value
```

Comment:
`TypeRestriction`'s semantics looks like `Cast`, so I think it is good that these getters rename same as `Cast`.